Source/ARTRealtimePresence.m

```diff
@@ -74,7 +75,16 @@ - (void)history:(void (^)(__GENERIC(ARTPaginatedResult, ARTPresenceMessage *) *,
 }
 
 - (BOOL)history:(ARTRealtimeHistoryQuery *)query callback:(void (^)(__GENERIC(ARTPaginatedResult, ARTPresenceMessage *) *, ARTErrorInfo *))callback error:(NSError **)errorPtr {
-    return [super history:query callback:callback error:errorPtr];
+    query.realtimeChannel = self.channel;
+    @try {
+        return [super history:query callback:callback error:errorPtr];
```

Comment:
Yes, [this](https://github.com/ably/ably-ios/blob/master/Source/ARTDataQuery.m#L58) could happen.